internal/dockerutil/setup.go

```diff
@@ -51,29 +52,57 @@ func DockerSetup(t *testing.T) (*dockertest.Pool, string) {
 }
 
 // dockerCleanup will clean up Docker containers, networks, and the other various config files generated in testing
-func dockerCleanup(testName string, pool *dockertest.Pool) func() {
+func dockerCleanup(t *testing.T, pool *dockertest.Pool) func() {
 	return func() {
 		cont, _ := pool.Client.ListContainers(docker.ListContainersOptions{All: true})
 		for _, c := range cont {
 			for k, v := range c.Labels {
-				if k == CleanupLabel && v == testName {
-					_ = pool.Client.StopContainer(c.ID, 10)
+				if k == CleanupLabel && v == t.Name() {
+					if err := pool.Client.StopContainer(c.ID, 10); isLoggableStopError(err) {
+						t.Logf("Failed to stop container %s during docker cleanup: %v", c.ID, err)
+					}
 					ctxWait, cancelWait := context.WithTimeout(context.Background(), time.Second*5)
-					_, _ = pool.Client.WaitContainerWithContext(c.ID, ctxWait)
-					_ = pool.Client.RemoveContainer(docker.RemoveContainerOptions{ID: c.ID, Force: true})
+					if _, err := pool.Client.WaitContainerWithContext(c.ID, ctxWait); err != nil {
+						t.Logf("Failed to wait for container %s during docker cleanup: %v", c.ID, err)
+					}
+					if err := pool.Client.RemoveContainer(docker.RemoveContainerOptions{
+						ID:            c.ID,
+						Force:         true,
+						RemoveVolumes: true}); err != nil {
```

Comment:
(*dockertest.Pool).Prune sets the RemoveVolumes to true. Previously, throughout the codebase, we were omitting this field. I opted to mimic what the package uses. 

It's subtle which is why I wanted to point this out. 🤞 maybe it helps with docker hanging issues.

Reply:
I couldn't find a quick answer on what remove volumes does, if it is a volume in use by other containers. I assume it silently leaves the volume in place. Removing the volume makes sense to me.